[FEAT] 채팅방 정보 조회 API 구현

dye0p opened the pull request:
## Issue Number
closed #824

## As-Is
<!-- 문제 상황 정의 -->
 현재 애플리케이션에는 채팅방 생성 기능만 존재하며, 생성된 채팅방의 상세 정보를 조회할 수 있는 기능이 부재한 상황입니다.

## To-Be
<!-- 변경 사항 -->
채팅방 정보 조회 기능을 추가하고, 안정적인 서비스 제공을 위해 유효성 검사 및 예외 처리 로직을 보강했습니다. 주요 변경 사항은 아래와 같습니다.

### 1. 채팅방 정보 조회 API 추가
   - GET /chatrooms/{chatroomId} 엔드포인트를 통해 채팅방 상세 정보를 조회하는 기능을 구현했습니다.
   - 응답 데이터에는 채팅 상대방 정보, 멘토링 정보 등이 포함되어 클라이언트의 화면 구성에 필요한 데이터를 제공합니다.

### 2. 유효성 검사 및 보안 강화
- 인가(Authorization): 채팅방 참여자(멘토, 멘티)만 정보를 조회할 수 있도록 인가 로직이 적용되었습니다. 비인가 접근 시 UnauthorizedChatRoomAccessException이 발생합니다.
- 예약 상태 확인: 예약 상태가 APPROVED(승인) 또는 COMPLETED(완료)인 경우에만 정보를 조회할 수 있도록 제한했습니다.
- 중복 생성 방지: 동일한 예약 건에 대해 채팅방이 중복 생성되는 것을 방지하기 위해 ChatRoomAlreadyExistsException 예외 처리 로직을 추가했습니다.

### 3. 예외 처리 구조 개선
- ChatRoomNotFoundException, ChatRoomAlreadyExistsException 등 특정 상황에 맞는 커스텀 예외를 정의하여 에러 상황을 명확히 구분하도록 개선했습니다.
- 신규 예외 처리를 위해 GlobalExceptionHandler를 업데이트했습니다.

### 4. 테스트 코드 추가
- 기능의 신뢰성을 확보하기 위해 ChatRoomServiceTest에 단위 테스트를 추가했습니다.
- 테스트는 정상 케이스와 함께 권한 없는 접근, 잘못된 예약 상태 등 다양한 예외 시나리오를 포함합니다.

## Check List
- [x] 테스트가 전부 통과되었나요?
- [x] 모든 commit이 push 되었나요?
- [x] merge할 branch를 확인했나요?
- [x] Assignee를 지정했나요?
- [x] Label을 지정했나요?
- [x] 닫을 이슈 번호를 지정했나요?

## (Optional) Additional Description
- 본 기능은 향후 구현될 실시간 채팅 기능의 기반 요소로, 클라이언트가 채팅 세션에 필요한 데이터를 조회할 수 있는 API를 제공하는 데 목적이 있습니다.
- 해당 PR의 브랜치는 앞선 [feature/811-create-chatroom](https://github.com/woowacourse-teams/2025-Fit-toring/pull/823) 브랜치에서 파생되었습니다.

<!-- This is an auto-generated comment: release notes by coderabbit.ai -->
## Summary by CodeRabbit

- New Features
  - 로그인 API가 사용자 ID와 토큰 정보를 함께 반환하여 로그인 직후 식별 가능.
  - 채팅방 조회 API 추가: 인증된 사용자가 자신의 채팅방 정보를 확인.
  - 채팅방 상태 관리 도입 및 접근 검증 강화(승인/완료 예약만 접근), 중복 생성 방지.
  - 채팅 관련 오류에 대해 적절한 상태코드/메시지로 응답.
- Tests
  - 채팅방 서비스/모델 및 인증 서비스 단위 테스트 추가.
- Chores
  - DB 마이그레이션: chat_room 테이블에 status 컬럼 추가.
<!-- end of auto-generated comment: release notes by coderabbit.ai -->

dye0p committed:
feat: 채팅방 관련 예외 클래스 및 핸들러 추가

채팅방 기능 구현에 필요한 커스텀 예외 클래스를 추가했음. 각 상황에 맞는 처리를 위해 예외를 세분화할 필요가 있었기 때문.

- `ChatRoomAlreadyExistsException`: 채팅방이 이미 존재할 경우 발생

- `ChatRoomNotFoundException`: 채팅방을 찾을 수 없는 경우 발생

- `UnauthorizedChatRoomAccessException`: 채팅방 접근 권한이 없는 경우 발생

추가된 예외들을 처리하기 위해 `GlobalExceptionHandler`에 핸들러 메서드를 추가함.

dye0p committed:
feat: 채팅방 조회 기능 추가

- `ChatRoomService`에 채팅방 조회 메서드 추가 (`findChatRoom`)
- 채팅방 존재 여부 및 참여자 권한 검증 로직 구현
- 예약 상태 확인을 통해 접근 가능 여부 검증 (`isChatRoomAccessibleStatus`)
- 멤버 정보와 멘토링 정보로 `ChatRoomResponse` 생성
- `Member`, `Reservation`, `ChatRoom`에 필요한 유틸 메서드(`isMentee`, `hasParticipant`) 추가
- `ChatRoomResponse` 클래스에 정적 팩토리 메서드 및 프로필 이미지 URL 처리 로직 추가

dye0p committed:
feat: 채팅방 단건 조회 API 컨트롤러 추가

- `@GetMapping`으로 `/chatrooms/{chatroomId}` 엔드포인트 정의
- 채팅방 조회 시 인증 필요하도록 `@AuthRequired` 추가
- `ChatRoomService.findChatRoom` 호출로 채팅방 정보 조회 및 응답
- 로그인 정보(`LoginInfo.memberId`)와 PathVariable(`chatroomId`)로 조회
- 응답 데이터로 `ChatRoomResponse` 반환

dye0p committed:
feat: 채팅방 생성 로직 보완 및 예약 검증 로직 추가

- `ChatRoomService.registerChatRoom`에 예약 및 멘토링 검증 메서드 추가
- `validateMentoring`: 존재하지 않는 멘토링 예외 처리 추가
- `validateReservationExists`: 이미 생성된 채팅방인지 확인 후 예외 처리
- `ChatRoomRepository`에 `existsByReservationId` 메서드 추가 (예약 기반 중복 확인)

dye0p committed:
test: ChatRoomService 테스트 추가

- 채팅방 생성(registerChatRoom) 성공/실패 케이스 테스트 추가
- 멘티/멘토의 채팅방 조회 성공/실패 케이스 테스트 추가
- 예약 상태 및 소프트 삭제된 멘토링 검증 테스트 추가
- 프로필 이미지 유효성 확인 테스트 추가

dye0p committed:
test: 채팅방 접근 예외 메시지 수정 검증 테스트 추가

- UnauthorizedChatRoomAccessException 메시지를 INVALID_STATUS_CHAT_ROOM_ACCESS로 변경 후 검증
- BusinessErrorMessage 변경 사항에 따른 테스트 수정

dye0p committed:
refactor: 컨벤션 수정

dye0p commented:
@CodeRabbit 리뷰해

kimdevv commented on backend/fittoring/src/main/java/fittoring/mentoring/business/service/ChatRoomService.java:
```diff
@@ -25,4 +50,92 @@ public ChatRoomCreatedInfo registerChatRoom(Reservation reservation) {
         return new ChatRoomCreatedInfo(url);
     }
+
+    private void validateMentoring(Mentoring mentoring) {
```
p2: 아래와 같이 `validateMentoringExists()`로 메서드명을 통일해 주는 건 어떨까요?

dye0p replied:
좋아요 수정했습니다!

kimdevv commented on backend/fittoring/src/main/java/fittoring/mentoring/business/service/ChatRoomService.java:
```diff
@@ -25,4 +50,92 @@ public ChatRoomCreatedInfo registerChatRoom(Reservation reservation) {
+                );
+    }
+
+    private void validateReservationExists(final Reservation reservation) {
```
p2: `final` 키워드는 붙이지 않기로 했던 것 같아서 빼도 될 것 같습니다!

kimdevv commented on backend/fittoring/src/main/java/fittoring/mentoring/business/service/ChatRoomService.java:
```diff
@@ -25,4 +50,92 @@ public ChatRoomCreatedInfo registerChatRoom(Reservation reservation) {
+
+        Member member = getMember(memberId);
+
+        String participant = getParticipantName(member, reservation);
```
p2: `participantName`이라고 하니 뭔가 자신의 이름이랑 혼동될 수 있는 것 같아요. **상대방의 이름**이라는 것을 명확하게 알 수 있도록 `getOpponentName()`이나 `getPartnerName()` 등은 어떤가요? 변수명도 동일하게요!

dye0p replied:
좋아요! 수정했습니다~

kimdevv commented on backend/fittoring/src/main/java/fittoring/mentoring/business/service/ChatRoomService.java:
```diff
@@ -25,4 +50,92 @@ public ChatRoomCreatedInfo registerChatRoom(Reservation reservation) {
+
+        String participant = getParticipantName(member, reservation);
+        String mentorName = reservation.getMentoring().getMentorName();
+        Mentoring mentoring = reservation.getMentoring();
```
p1: `mentoring`을 먼저 가져오고 나면 `reservation.getMentoring()`을 두 번 호출할 필요가 없겠군요.

dye0p replied:
이 부분은 제거했습니다! 왜냐하면 멘토링의 정보를 가져오는 책임은 `MentoringService`에서 수행해야 한다고 판단했어요. `ChatRoomService`는 채팅방의 정보만 가져와야 한다고 생각해서 제거했습니다!

kimdevv commented on backend/fittoring/src/main/java/fittoring/mentoring/business/exception/BusinessErrorMessage.java:
```diff
@@ -41,7 +41,10 @@ public enum BusinessErrorMessage {
     NOT_CERTIFICATE_OWNER("자신의 자격사항이 아닙니다."),
     MEMBER_ROLE_NOT_FOUND("존재하지 않는 사용자 권한입니다. "),
-    ;
+    CHAT_ROOM_NOT_FOUNT("존재하지 않는 채팅방 입니다."),
```
p1: 토끼가 하는 말처럼 오타 수정해 주시면 좋을 것 같네요~

dye0p replied:
수정했습니다!

kimdevv reviewed (changes requested):
고생하셨습니다~ 간단한 코멘트 남겼으니 확인 한 번씩 부탁드려요!

dye0p committed:
refactor: 잘못된 패키지 위치 이동

dye0p committed:
refactor: 불필요한 final 키워드 제거

dye0p committed:
feat: 채팅방 엔티티 채팅 상태 필드 추가

- flyway chat_room 테이블 status 칼럼 추가 스크립트 추가 (VARCHAR, NOT NULL)
- 채팅방의 상태 관리 필요성을 느껴 필드 추가

dye0p committed:
feat: 로그인 응답에 회원 식별자 추가

- `LoginResponse` DTO 생성 (memberId와 AuthToken 포함)
- AuthService의 `login` 메서드 반환 타입 변경 (`AuthTokenResponse` → `LoginResponse`)
- AuthController의 `/login` API 응답 변경 (회원 식별자 반환 추가)
- 관련 테스트 수정 및 정상 응답 데이터 검증 항목 추가

dye0p committed:
refactor: ChatRoomResponse 응답 필드 수정

- 기존 ChatRoomResponse에 포함되어있던 멘토링 관련 필드를 제거하였음
- 멘토링의 정보는 별도의 API로 관리하는 것이 관심사를 분리하는 것에 적절하다 판단하였음
- "상대방"이라는 뜻을 나타내기 위해서 participant 네이밍을 사용했지만, opponent이 더 적절하다 판단하여 변경하였음

dye0p commented:
@CodeRabbit 다시 리뷰해

dye0p committed:
fix: 잘못된 채팅방 예외 메시지 수정

dye0p committed:
refactor: ChatRoomAlreadyExistsException http 상태코드 변경

dye0p commented on backend/fittoring/src/main/java/fittoring/application/exception/ChatRoomAlreadyExistsException.java:
```diff
@@ -0,0 +1,8 @@
+package fittoring.mentoring.business.exception;
+
+public class ChatRoomAlreadyExistsException extends RuntimeException {
+
+    public ChatRoomAlreadyExistsException(String message) {
+        super(message);
+    }
+}
```
적용했어 고마워!

dye0p committed:
refactor: ChatRoom 엔티티 EqualsAndHashCode 적용

- id 필드를 @EqualsAndHashCode.Include로 설정

dye0p commented on backend/fittoring/src/main/java/fittoring/domain/model/ChatRoom.java:
```diff
@@ -52,7 +60,32 @@ public class ChatRoom {
+        );
+    }
+
+    public boolean hasParticipant(Long memberId) {
+        return mentorId.equals(memberId) || menteeId.equals(memberId);
     }
```
이건 괜찮을 거 같아!

dye0p committed:
refactor: Reservation 채팅방 접근 상태 검증 로직 개선

- 메서드 네이밍 직관적으로 변경
- 상태 검증 로직을 기존의 메서드를 재사용 하도록 변경

dye0p committed:
refactor: Reservation 메서드 배치 순서 변경

dye0p commented on backend/fittoring/src/main/java/fittoring/mentoring/business/model/Reservation.java:
```diff
@@ -112,4 +112,8 @@ public String getMentorPhone() {
+
+    public boolean isChatRoomAccessibleStatus() {
+        return status.isApprove() || status.isComplete();
+    }
```
메서드 네이밍 추천은 고마워. 하지만 채팅방 접근 규칙을 분리하는건 아직 이른 것 같아.

dye0p committed:
refactor: ChatRoom 엔티티 reservationId 필드에 유니크 제약조건 추가

dye0p commented on backend/fittoring/src/main/java/fittoring/application/mentoring/repository/ChatRoomRepository.java:
```diff
@@ -4,4 +4,6 @@
 
 public interface ChatRoomRepository extends ListCrudRepository<ChatRoom, Long> {
+
+    boolean existsByReservationId(Long reservationId);
```
추가했어!

dye0p committed:
refactor: 멘토링 존재 검증 메서드 네이밍 명확하게 변경

dye0p reviewed (commented):
남겨주신 코멘트를 반영했습니다! 다시 리뷰 부탁드려요!

dye0p committed:
test: 예약 상태에 따른 채팅방 조회 테스트 완료 경우 추가 및 ParameterizedTest로 테스트 통합

dye0p commented on backend/fittoring/src/test/java/fittoring/application/chatroom/service/ChatRoomServiceTest.java:
```diff
@@ -0,0 +1,321 @@
+        em.persist(mentee);
+
+        Reservation reservation = new Reservation("대기중 예약", Status.PENDING, mentoring, mentee);
+        em.persist(reservation);
+
+        ChatRoom chatRoom = new ChatRoom(reservation.getId(), mentee.getId(), mentor.getId());
+        em.persist(chatRoom);
+
+        // when
+        // then
+        assertThatThrownBy(() -> chatRoomService.findChatRoom(mentee.getId(), chatRoom.getId()))
+                .isInstanceOf(UnauthorizedChatRoomAccessException.class)
+                .hasMessage(BusinessErrorMessage.INVALID_STATUS_CHAT_ROOM_ACCESS.getMessage());
+    }
+
+    @DisplayName("채팅방 조회시 승인(APPROVED), 완료(COMPLETED) 상태의 예약은 조회 가능하다")
+    @Test
+    void findChatRoomApprovedOrCompletedReservationStatus() {
+        // given
+        Member mentor = new Member("id7", "MALE", "멘토", new Phone("010-1234-0000"), Password.from("pw"));
+        em.persist(mentor);
+
+        Mentoring mentoring = new Mentoring(mentor, 12000, 2, "컨텐츠", "소개", "링크");
+        em.persist(mentoring);
+
+        Member mentee = new Member("mentee7", "MALE", "멘티", new Phone("010-0000-1111"), Password.from("pw2"));
+        em.persist(mentee);
+
+        Reservation approvedReservation = new Reservation("승인된 예약", Status.APPROVED, mentoring, mentee);
+        em.persist(approvedReservation);
+
+        ChatRoom chatRoom = new ChatRoom(approvedReservation.getId(), mentee.getId(), mentor.getId());
+        em.persist(chatRoom);
+
+        // when
+        ChatRoomResponse response = chatRoomService.findChatRoom(mentee.getId(), chatRoom.getId());
+
+        // then
+        assertThat(response).isNotNull();
+    }
```
추가했어!

dye0p committed:
refactor: ChatRoomResponse 네이밍 변경

dye0p committed:
feat: 채팅방 멘토링 정보 조회 기능 추가

- 필요한 컬럼만 select 할 수 있도록 dtoProjection 사용

dye0p committed:
feat: 채팅방 정보 조회를 위한 ChatRoomFacadeService 추가

- 멘토링 정보, 채팅방 정보를 각각의 service에서 호출하여 ChatRoomInfoResponse 생성하기 위함

dye0p committed:
test: ChatRoomIntegrationTest 추가

dye0p committed:
feat: 채팅방 정보 조회 응답 dto에 myRole 필드 추가

dye0p committed:
refactor: 채팅방 정보 조회 응답 dto 채팅방 상태 필드 타입 Enum타입으로 변경

- Enum 타입으로 선언하여 다른 값이 못들어가게 할 수 있음

kimdevv commented on backend/fittoring/src/main/java/fittoring/exception/GlobalExceptionHandler.java:
```diff
@@ -201,6 +204,21 @@ public ResponseEntity<ErrorResponse> handle(MissingRequestCookieException e) {
+
+    @ExceptionHandler(ChatRoomAlreadyExistsException.class)
+    public ResponseEntity<ErrorResponse> handle(ChatRoomAlreadyExistsException e) {
+        return buildErrorResponse(e, HttpStatus.CONFLICT, e.getMessage());
```
p3: `CONFLICT`라는 응답 코드가 등장했네요

dye0p replied:
중복 상태를 나타내는 http 응답이라고 합니다~
[409 Conflict - HTTP - MDN](https://developer.mozilla.org/ko/docs/Web/HTTP/Reference/Status/409)

kimdevv commented on backend/fittoring/src/main/java/fittoring/domain/model/Reservation.java:
```diff
@@ -64,29 +64,33 @@ public Reservation(String content, Status status, Mentoring mentoring, Member me
+    }
+
+    public void changeStatus(Status updateStatus) {
+        this.status.validateReservation(updateStatus);
```
p2: `Status`는 상태를 나타내는 값의 집합인데 `Reservation`의 상태를 검증하는 역할이 있어도 괜찮을까요? 예약의 상태를 검증하는 로직은 예약이 가져도 된다고 생각합니다!

dye0p replied:
처음 구현할 땐 상태를 검증하는 것이므로 status가 하는 것이 좋겠다고 생각했습니다. 하지만, 다시 생각해보니 이것은 예약의 상태를 검증하는거니 말씀해주신대로 예약에서 책임을 가져야 될 것 같네요! 만약 또 다른 도메인의 상태를 검증할 때 지금처럼 status에 둔다면, status의 책임이 늘어날 것 같아요! 코멘트 감사합니다👍

kimdevv commented on backend/fittoring/src/main/java/fittoring/application/mentoring/repository/MentoringRepository.java:
```diff
@@ -21,6 +22,21 @@ public interface MentoringRepository extends ListCrudRepository<Mentoring, Long>
 
+    @Query("""
+            SELECT new fittoring.mentoring.business.service.dto.chat.ChatRoomMentoringInfoDto(
+                m.mentor.name,
```
p2: `Member` 테이블을 조인해주면 `m.mentor`을 가져올 때 생기는 N+1을 예방할 수 있을 것 같아요!

dye0p replied:
좋습니다! 수정했습니다~

ppparkta commented on backend/fittoring/src/main/java/fittoring/domain/model/ChatRoom.java:
```diff
@@ -3,20 +3,24 @@
 import org.springframework.data.annotation.CreatedDate;
 import org.springframework.data.jpa.domain.support.AuditingEntityListener;
 
+@EqualsAndHashCode(onlyExplicitlyIncluded = true)
```
p3: 오 처음 알았어요! @EqualsAndHashCode.Include랑 같이 쓰이는건가보네요!

ppparkta commented on backend/fittoring/src/main/java/fittoring/domain/model/ChatRoom.java:
```diff
@@ -52,7 +63,32 @@ public class ChatRoom {
+                false,
+                null
+        );
+    }
```
p3: 두 생성자의 개행 방법을 통일시켜주면 좋을 것 같아요!

dye0p replied:
테스트를 위해서 모든 매개변수를 받도록 생성자를 열어뒀는데, 필요없을 것 같아서 지웠어요!

jye326 commented on backend/fittoring/src/main/java/fittoring/mentoring/presentation/dto/ChatRoomInfoDto.java:
```diff
@@ -0,0 +1,12 @@
+
+public record ChatRoomInfoDto(
+        Long mentoringId,
+        MemberRole myRole,
```
p3 : `MemberRole`은 왜 필요한건가요?

dye0p replied:
멘토와 멘티별로 같은 채팅방에서 필요한 버튼과 정보들이 달라지기 때문입니다!

**멘토가 보는 화면(피그마)**
<img width="259" height="117" alt="image" src="https://github.com/user-attachments/assets/2041baab-303f-40f4-b4b5-d72e9e4c1b5d" />

**멘티가 보는 화면(피그마)**
<img width="259" height="117" alt="image" src="https://github.com/user-attachments/assets/66cdaea4-0c63-4471-b4d0-1c4f6a796cb2" />

MemberRole를 함께 응답에 포함시켜 member의 권한을 클라이언트쪽에서 체크할 수 있습니다!

kimdevv reviewed (commented):
고생하셨습니다~ 간단한 커멘트 남깁니다!

jye326 reviewed (approved):
구현 잘 해주셨네요! 궁금한 점 코멘트 남기고 승인할게요~

ppparkta commented on backend/fittoring/src/main/java/fittoring/application/mentoring/repository/MentoringRepository.java:
```diff
@@ -21,6 +22,21 @@ public interface MentoringRepository extends ListCrudRepository<Mentoring, Long>
             """)
     Optional<Mentoring> findByReviewId(Long reviewId);
 
+    @Query("""
+            SELECT new fittoring.mentoring.business.service.dto.chat.ChatRoomMentoringInfoDto(
+                m.mentor.name,
+                m.price,
+                (SELECT img.url
+                   FROM Image img
+                  WHERE img.relationId = m.id
+                    AND img.imageType = 'MENTORING_PROFILE'
+                    AND img.imageVariant = 'THUMBNAIL')
+            )
+            FROM Mentoring m
+            WHERE m.id = :mentoringId
+            """)
+    Optional<ChatRoomMentoringInfoDto> findByIdForChatRoom(Long mentoringId);
```
p2: 이미지 저장 시 누락 등으로 인해 THUMBNAIL이 없을 가능성이 있는데 이 때 대신 'DEFAULT' variant의 이미지를 가지고 올 수 있도록 추가해주면 좋을 것 같아요!

```java
@Query("""
SELECT new fittoring.mentoring.business.service.dto.chat.ChatRoomMentoringInfoDto(
    m.mentor.name,
    m.price,
    COALESCE(
        (SELECT img.url
           FROM Image img
          WHERE img.relationId = m.id
            AND img.imageType = 'MENTORING_PROFILE'
            AND img.imageVariant = 'THUMBNAIL'),
        (SELECT img.url
           FROM Image img
          WHERE img.relationId = m.id
            AND img.imageType = 'MENTORING_PROFILE'
            AND img.imageVariant = 'DEFAULT')
    )
)
FROM Mentoring m
WHERE m.id = :mentoringId
""")
Optional<ChatRoomMentoringInfoDto> findByIdForChatRoom(Long mentoringId);
```

dye0p replied:
그렇군요! 올려주신 쿼리로 수정하겠습니다!

ppparkta commented on backend/fittoring/src/main/java/fittoring/application/mentoring/service/ChatRoomFacadeService.java:
```diff
@@ -0,0 +1,22 @@
+    private final ChatRoomService chatRoomService;
+
+    public ChatRoomInfoResponse getChatRoom(Long chatroomId, Long memberId) {
+        ChatRoomInfoDto chatRoomInfo = chatRoomService.findChatRoom(chatroomId, memberId);
+        ChatRoomMentoringInfoDto mentoringInfo = mentoringService.findMentoringInfoForChatRoom(
+                chatRoomInfo.mentoringId()
+        );
```
p3: 퍼싸드 적용하니 읽기에도 깔끔하고 각 서비스가 자신이 담당하는 도메인 관련된 기능만 수행할 수 있어서 유지보수하기도 좋아보이네요 👍

ppparkta commented on backend/fittoring/src/main/java/fittoring/mentoring/business/service/ChatRoomInfoResponse.java:
```diff
@@ -0,0 +1,10 @@
+        ChatRoomMentoringInfoDto mentoringInfoDto,
+        ChatRoomInfoDto chatRoomInfoDto
+) {
+}
```
p2: 컨트롤러 계층에서 서비스 계층에 쓰이는 dto를 그대로 사용하게 되면 레이어간 결합도가 높아져서 컨트롤러 응답 조건이 달라졌을 때 서비스 레이어의 dto까지 수정해야 하는 문제가 있을 것 같아요.

필드로 서비스 레이어의 dto를 직접 갖는 것보다 원시 값을 직접 갖고 정적 팩토리 메서드 등으로 dto->response 변환 로직을 수행해주는거 어떠신가요?

dye0p replied:
그렇네요! 수정하겠습니다~

ppparkta commented on backend/fittoring/src/main/resources/db/migration/V202510081528__add_unique_constraint_on_chat_room.sql:
p3: 같은 테이블의 제약조건은 하나의 파일로 합치는게 보기 더 편할 것 같아요. 수정 시점이 다르면 어쩔 수 없지만 아직 develop에 머지되기 전이니 의견내봅니다!

dye0p replied:
좋습니다~

ppparkta reviewed (commented):
고생하셨습니다. 서비스 코드가 깔끔해서 정말 좋네요! 

멘토링 썸네일 조회 부분은 수정이 꼭 필요해보여서 코멘트로 남깁니다. 이미지 부분만 수정 후 재요청 주시면 바로 승인하겠습니다.

dye0p committed:
Merge branch 'develop' into feature/824-get-chat-room-info

# Conflicts:
#	backend/fittoring/src/main/java/fittoring/application/chatroom/service/ChatRoomService.java
#	backend/fittoring/src/main/java/fittoring/application/exception/BusinessErrorMessage.java
#	backend/fittoring/src/main/java/fittoring/application/exception/ChatRoomAlreadyExistsException.java
#	backend/fittoring/src/main/java/fittoring/application/exception/ChatRoomNotFoundException.java
#	backend/fittoring/src/main/java/fittoring/application/exception/UnauthorizedChatRoomAccessException.java
#	backend/fittoring/src/main/java/fittoring/application/mentoring/presentation/ChatRoomController.java
#	backend/fittoring/src/main/java/fittoring/application/mentoring/service/ChatRoomFacadeService.java
#	backend/fittoring/src/main/java/fittoring/application/mentoring/service/ChatRoomInfoResponse.java
#	backend/fittoring/src/main/java/fittoring/application/mentoring/service/MentoringService.java
#	backend/fittoring/src/main/java/fittoring/application/mentoring/service/dto/chat/ChatRoomMentoringInfoDto.java
#	backend/fittoring/src/main/java/fittoring/domain/model/ChatStatus.java
#	backend/fittoring/src/main/java/fittoring/exception/GlobalExceptionHandler.java
#	backend/fittoring/src/main/java/fittoring/mentoring/business/repository/MentoringRepository.java
#	backend/fittoring/src/main/java/fittoring/mentoring/business/service/AuthService.java
#	backend/fittoring/src/main/java/fittoring/mentoring/presentation/api/AuthController.java
#	backend/fittoring/src/test/java/fittoring/application/auth/service/AuthServiceTest.java
#	backend/fittoring/src/test/java/fittoring/domain/model/ChatRoomTest.java
#	backend/fittoring/src/test/java/fittoring/mentoring/business/service/MentoringServiceTest.java

dye0p committed:
refactor: 예약 상태 검증 로직 이동(status -> reservation)

dye0p committed:
feat: Member를 가져올 수 있도록 JOIN 추가

dye0p committed:
refactor: 채팅방 정보 조회 응답 필드가 원시값을 가지도록 변경

- 기존 필드로 원시값을 가지던 방식에서 원시 필드를 가지도록 변경
- 변경에 유연하게 대응하기 위해서 변경

dye0p committed:
refactor: 사용하지 않는 생성자 제거

dye0p committed:
refactor: 채팅방 스키마 수정 관련 스크립트 하나의 스크립트로 이동

- 따로 존재했던 채팅방 스키마 수정 스크립트를 하나의 채팅방 생성 스키마로 이동하고, 이동된 스크립트는 제거하였음

dye0p committed:
refactor: 멘토링 프로필 썸네일이 없는 경우 기본 이미지를 반환하도록 쿼리 수정

dye0p reviewed (commented):
리뷰 반영했으니 확인해주세요!

dye0p committed:
Merge branch 'develop' into feature/824-get-chat-room-info

# Conflicts:
#	backend/fittoring/src/main/java/fittoring/application/chatroom/service/ChatRoomService.java
#	backend/fittoring/src/main/java/fittoring/application/exception/BusinessErrorMessage.java
#	backend/fittoring/src/main/java/fittoring/application/exception/ChatRoomAlreadyExistsException.java
#	backend/fittoring/src/main/java/fittoring/application/exception/ChatRoomNotFoundException.java
#	backend/fittoring/src/main/java/fittoring/application/exception/UnauthorizedChatRoomAccessException.java
#	backend/fittoring/src/main/java/fittoring/application/mentoring/presentation/ChatRoomController.java
#	backend/fittoring/src/main/java/fittoring/application/mentoring/service/ChatRoomFacadeService.java
#	backend/fittoring/src/main/java/fittoring/application/mentoring/service/ChatRoomInfoResponse.java
#	backend/fittoring/src/main/java/fittoring/application/mentoring/service/MentoringService.java
#	backend/fittoring/src/main/java/fittoring/application/mentoring/service/dto/chat/ChatRoomMentoringInfoDto.java
#	backend/fittoring/src/main/java/fittoring/domain/model/ChatStatus.java
#	backend/fittoring/src/main/java/fittoring/exception/GlobalExceptionHandler.java
#	backend/fittoring/src/main/java/fittoring/mentoring/business/repository/MentoringRepository.java
#	backend/fittoring/src/main/java/fittoring/mentoring/business/service/AuthService.java
#	backend/fittoring/src/main/java/fittoring/mentoring/presentation/api/AuthController.java
#	backend/fittoring/src/test/java/fittoring/application/auth/service/AuthServiceTest.java
#	backend/fittoring/src/test/java/fittoring/domain/model/ChatRoomTest.java
#	backend/fittoring/src/test/java/fittoring/mentoring/business/service/MentoringServiceTest.java

dye0p committed:
test: MentoringIntegrationTest에 DirtiesContext 추가

- @DirtiesContext(classMode = ClassMode.AFTER_CLASS)로 테스트 환경 초기화 설정 추가

dye0p committed:
Merge branch 'develop' into feature/824-get-chat-room-info
# Please enter a commit message to explain why this merge is necessary,
# especially if it merges an updated upstream into a topic branch.
#
# Lines starting with '#' will be ignored, and an empty message aborts
# the commit.
